### DISCUSSION TITLE
Column layout with fixed height and dynamic content

### DISCUSSION THREAD

**Discussion by luudv** [facebook/lexical#5405] [Q&A] [answered]
Columns layout is a useful plugin. The problem I pose is whether it is possible to have a Columns layout with a fixed height, the nodes inside each column will automatically jump to the next column when the current column is filled?
<img width="1218" alt="2023-12-19_3-27-16 PM" src="https://github.com/facebook/lexical/assets/2778871/7a48ff0f-1982-4005-8a69-a937e938d745">

**Comment by ivailop7** [comment, marked as answer]
The layout plugin in the Playground is just an example, not production-ready. You can base your implementation off it. You can set the CSS applied to the layout item to have a fixed height, for the jump over logic, you will need to change the plugin doing some kind of a tracking varible keeping track of if you've reached the paragraph end and move the selection to the next box. An example would be to keep track of the box height from the DOM, the moment the height is > 300px to move selection. Overly simplistic example, but that's the premise. Worth looking into Node Transforms as well in the docs.

**Reply by luudv** [reply]
Thanks for your advice,

I'm also solving a similar problem, the algorithm I'm doing is comparing the height of the input content and the height of each column and automatically pushing the content to the new column with a recursive method.

However, the problem I am facing is that the virtual doms are not generated when I use functions like **setTextContent** (for **textNode**). Therefore I cannot calculate the height of the content each time using these functions. Do you have any advice for me?
